zebra/irdp_main.c

```diff
@@ -234,7 +234,8 @@ int irdp_send_thread(struct thread *t_advert)
       }
 
   tmp = irdp->MaxAdvertInterval-irdp->MinAdvertInterval;
-  timer =  (random () % tmp ) + 1;
+  assert (tmp > 0);
+  timer = (random () % tmp) + 1;
   timer = irdp->MinAdvertInterval + timer;
```

Comment:
You cited the [v6 version](https://tools.ietf.org/html/rfc4861), but the [v4 version](https://tools.ietf.org/html/rfc1256) is what is under consideration. The latter states:

```
   MaxAdvertisementInterval
                 The maximum time allowed between sending multicast
                 Router Advertisements from the interface, in seconds.
                 Must be no less than 4 seconds and no greater than 1800
                 seconds.

                 Default: 600 seconds

   MinAdvertisementInterval
                 The minimum time allowed between sending unsolicited
                 multicast Router Advertisements from the interface, in
                 seconds.  Must be no less than 3 seconds and no greater
                 than MaxAdvertisementInterval.

                 Default: 0.75 * MaxAdvertisementInterval
```
I read this as `MinAdvertisementInterval <= MaxAdvertisementInterval` and implemented it thusly the first go-round; I was advised that the RFC authors perhaps intended `<` and not `<=` and revised my commit.

@donaldsharp 
@eqvinox